src/object_schema.cpp

```diff
@@ -214,6 +245,78 @@ static void validate_property(Schema const& schema,
 
 void ObjectSchema::validate(Schema const& schema, std::vector<ObjectSchemaValidationException>& exceptions) const
 {
+
```

Comment:
A rewrite of this which reduces the runtime from O(N^2) to O(N log N):

```c++
    std::vector<StringData> public_property_names;
    std::vector<StringData> internal_property_names;
    internal_property_names.reserve(persisted_properties.size() + computed_properties.size());
    auto gather_names = [&](auto const& properties) {
        for (auto const& prop : properties) {
            internal_property_names.push_back(prop.name);
            if (!prop.public_name.empty())
                public_property_names.push_back(prop.public_name);
        }
    };
    gather_names(persisted_properties);
    gather_names(computed_properties);
    std::sort(public_property_names.begin(), public_property_names.end());
    std::sort(internal_property_names.begin(), internal_property_names.end());

    // Check that property names and aliases are unique
    auto for_each_duplicate = [](auto&& container, auto&& fn) {
        auto end = container.end();
        for (auto it = std::adjacent_find(container.begin(), end); it != end; it = std::adjacent_find(it + 2, end))
            fn(*it);
    };
    for_each_duplicate(public_property_names, [&](auto property_name) {
        exceptions.emplace_back("Alias '%1' appears more than once in the schema for type '%2'.",
                                property_name, name);
    });
    for_each_duplicate(internal_property_names, [&](auto internal_name) {
        exceptions.emplace_back("Property '%1' appears more than once in the schema for type '%2'.",
                                internal_name, name);
    });

    // Check that no aliases conflict with property names
    struct ErrorWriter {
        ObjectSchema const& os;
        std::vector<ObjectSchemaValidationException>& exceptions;

        struct Proxy {
            ObjectSchema const& os;
            std::vector<ObjectSchemaValidationException>& exceptions;
            Proxy& operator=(StringData name) {
                exceptions.emplace_back("Property '%1.%2' has an alias '%3' that conflicts with a property of the same name.",
                                        os.name, os.property_for_public_name(name)->name, name);
                return *this;
            }
        };
        Proxy operator*() { return Proxy{os, exceptions}; }
        ErrorWriter& operator++() { return *this; }
        ErrorWriter& operator++(int) { return *this; }
    } writer{*this, exceptions};
    std::set_intersection(public_property_names.begin(), public_property_names.end(),
                          internal_property_names.begin(), internal_property_names.end(), writer);
```

This makes one of the tests fail due to adding the object type to the error message and changing the order that alias/name collisions are reported in.